File: rust/src/commands/generate.rs
use camino::Utf8PathBuf;
use clap::{Args, Subcommand};

#[derive(Args, Debug)]
pub struct GenerateArgs {
    #[command(subcommand)]
    pub command: GenerateCommand,

    /// Project root (default: current directory)
    #[arg(long, short = 'C')]
    pub root: Option<Utf8PathBuf>,

    /// Output file (default: stdout)
    #[arg(long, short)]
    pub output: Option<Utf8PathBuf>,
}

#[derive(Subcommand, Debug)]
pub enum GenerateCommand {
    /// Generate .env.example
    EnvExample,
    /// Generate ENVIRONMENT.md documentation
    EnvDoc,
    /// Generate TypeScript types (env.d.ts)
    EnvTypes,
    /// Generate JSON schema for configuration
    ConfigSchema,
    /// Generate TOML config template
    ConfigTemplate,
    /// Generate GitHub Actions workflow snippet
    GithubActions,
}

pub async fn generate(args: GenerateArgs) -> Result<u8, anyhow::Error> {
    let root = args.root.unwrap_or_else(|| Utf8PathBuf::from("."));
    let root = std::fs::canonicalize(&root)?
        .try_into()
        .map_err(|_| anyhow::anyhow!("Invalid root path"))?;

    let (model, config) = crate::core::pipeline::load_project(&root).await;

    let output = match args.command {
        GenerateCommand::EnvExample => {
            crate::generators::generate_env_example(&model, &config)
        }
        GenerateCommand::EnvDoc => {
            crate::generators::generate_environment_doc(&model, &config)
        }
        GenerateCommand::EnvTypes => {
            crate::generators::generate_env_types(&model, &config)
        }
        GenerateCommand::ConfigSchema => {
            crate::generators::generate_config_schema()
        }
        GenerateCommand::ConfigTemplate => {
            crate::generators::generate_config_template()
        }
        GenerateCommand::GithubActions => {
            crate::generators::generate_github_actions(&model, &config)
        }
    };

    if let Some(path) = &args.output {
        std::fs::write(path, output)?;
        println!("Written to {}", path);
    } else {
        println!("{}", output);
    }

    Ok(0)
}

File: dart/lib/src/commands/generate.dart
/// `envdoctor generate <target>` — generate files from the project model.
library;

import 'dart:io';

import '../core/pipeline.dart';
import '../generators/env_example.dart';
import '../generators/env_types.dart';
import '../generators/environment_doc.dart';
import '../generators/github_actions.dart';
import '../generators/schema.dart';
import '../utils/paths.dart';
import 'shared.dart';

enum GenerateTarget {
  envExample,
  envDoc,
  envTypes,
  configSchema,
  configTemplate,
  githubActions,
}

class GenerateOptions {
  final GenerateTarget target;
  final String rootDir;
  final String? output;

  GenerateOptions({required this.target, required this.rootDir, this.output});
}

int runGenerate(GenerateOptions opts) {
  final root = opts.rootDir;

  final context = loadProject(root);
  final model = context.model;
  final config = context.config;

  final output = switch (opts.target) {
    GenerateTarget.envExample => generateEnvExample(model, config),
    GenerateTarget.envDoc => generateEnvironmentDoc(model, config),
    GenerateTarget.envTypes => generateEnvTypes(model, config),
    GenerateTarget.configSchema => generateConfigSchema(),
    GenerateTarget.configTemplate => generateConfigTemplate(),
    GenerateTarget.githubActions => generateGithubActions(model, config),
  };

  if (opts.output != null) {
    final path =
        isAbsolutePath(opts.output!) ? opts.output! : joinPath(root, opts.output!);
    File(path).writeAsStringSync(output);
    out('Written to ${opts.output}\n');
  } else {
    out('$output\n');
  }

  return 0;
}
